tests/combine_samples.C

```diff
@@ -0,0 +1,200 @@
+//      [4] Optional: "-o" or "offset" to subtract an arbitrary offset vector
+//          before computing the SVD. Offset vector file supplied as next argument.
+//
+// Outputs:
+//	[1] total_snapshot.*:           single hdf5 file (per rank) of all loaded data.
+//	[2] mean.*:                     single hdf5 file (per rank) with subtracted mean.
+//	[3] total.* OR total_meansub.*: single hdf5 file (per rank) with the new POD basis.
+//
+// Examples: mpirun -n 2 ./combine_samples file1 file2 file3 -b -m
+//           mpirun -n 2 ./combine_samples -o offsetfile file1 file2 file3
+
+#include "linalg/BasisGenerator.h"
+#include "linalg/scalapack_wrapper.h"
+#include "linalg/BasisReader.h"
+#include "linalg/BasisWriter.h"
+#include <vector>
+#include "mpi.h"
+#include <string>
+#include <stdio.h>
+
+int main(int argc, char* argv[])
+{
+    // Initialize MPI
+    MPI_Init(&argc, &argv);
+    int size;
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+	
+    std::vector<std::string> sample_names;
+    int snaps = 0; 
+    int dim   = 0;
+    bool subtract_mean = false;
+    bool subtract_offset = false;
+    std::string kind = "snapshot";
+    std::string offset_file;
+
+    bool offset_arg = false;	
+
+    if (argc >= 2) {
+        for (int i = 1; i < argc; i++) {
+	    if (!strcmp(argv[i], "basis") || !strcmp(argv[i], "-b")) { 
+            if (rank==0) std::cout << "Argument " << i << " identified as basis or -b" << std::endl;
+                kind = "basis"; 
+            } 
+            else if (offset_arg) {
+                offset_file = argv[i];
+                offset_arg = false;
+            }
+            else if (!strcmp(argv[i], "offset") || !strcmp(argv[i], "-o")) {
+                if (rank==0) std::cout << "Will subtract offset" << std::endl;
+                subtract_offset = true;
+                offset_arg = true;
+            }
+            else if (!strcmp(argv[i], "mean") || !strcmp(argv[i], "-m")) {
+                if (rank==0) std::cout << "Will subtract mean" << std::endl;
+	    	subtract_mean = true;
+	    }
+	    else {
+                sample_names.push_back(argv[i]); 
+	    }
+	}
+    }
+    else {
+        if (rank==0) std::cout << "No arguments passed." << std::endl;
+        return 1;
+    }
+ 
+    /*-- Read dimension and count number of snapshots/bases --*/
+    if (rank==0) std::cout << "Opening files to read dimension and count number of snapshots/bases" << std::endl;
+    for (const auto& sample_name: sample_names) {
+        CAROM::BasisReader reader(sample_name);
+		
+        if (kind == "snapshot") {
+	    CAROM::Matrix *snapshots = (CAROM::Matrix*) reader.getSnapshotMatrix(0);
+	    dim    = snapshots->numRows();
+ 	    snaps += snapshots->numColumns();
+        }
+	else {
+            CAROM::Matrix *basis = (CAROM::Matrix*) reader.getSpatialBasis(0);
+	    dim    = basis->numRows();
+	    snaps += basis->numColumns();
+	}
+    }
```

Comment:
The verification of `dim > 0` could go here, along with setting another variable `dimFirst = dim` if `dimFirst` is not already set, otherwise verification that `dim == dimFirst`. This will ensure all files have the same dimensions, which could help users prevent or find mistakes.

Reply:
Added both verifications